dango/app/src/proposal_preparer.rs

```diff
@@ -0,0 +1,89 @@
+use {
+    dango_types::{
+        config::AppConfig,
+        oracle::{ExecuteMsg, PriceSource, QueryPriceSourcesRequest},
+    },
+    grug::{Binary, Coins, Json, JsonSerExt, Message, NonEmpty, QuerierWrapper, StdError, Tx},
+    prost::bytes::Bytes,
+    std::time,
+    thiserror::Error,
+};
+
+const PYTH_URL: &str = "https://hermes.pyth.network";
+const DEFAULT_TIMEOUT: time::Duration = time::Duration::from_millis(500);
+const GAS_LIMIT: u64 = 5_000_000;
+
+#[derive(Debug, Error)]
+pub enum ProposerError {
+    #[error(transparent)]
+    Std(#[from] StdError),
+
+    #[error(transparent)]
+    Reqwest(#[from] reqwest::Error),
+}
+
+impl From<ProposerError> for grug_app::AppError {
+    fn from(value: ProposerError) -> Self {
+        grug_app::AppError::PrepareProposal(value.to_string())
+    }
+}
+
+#[derive(Clone)]
+pub struct ProposalPreparer;
+
+impl grug_app::ProposalPreparer for ProposalPreparer {
+    type Error = ProposerError;
+
+    fn prepare_proposal(
+        &self,
+        querier: QuerierWrapper,
+        mut txs: Vec<Bytes>,
+        _max_tx_bytes: usize,
+    ) -> Result<Vec<Bytes>, Self::Error> {
+        let cfg: AppConfig = querier.query_app_config()?;
+
+        // Retrieve the price ids from the oracle and prepare the query params.
+        let params = querier
+            .query_wasm_smart(cfg.addresses.oracle, QueryPriceSourcesRequest {
+                start_after: None,
+                limit: Some(u32::MAX),
+            })?
+            .into_values()
+            .filter_map(|price_id| {
```

Comment:
`price_id` should be renamed to `PriceSource`, since it is of `PriceSource` type. `price_id` doesn't make sense...

Reply:
Yes, you're right